### PR TITLE
[Perldoc] Documentation for get_dicom_info.pl

### PR DESCRIPTION
**Pull request by cmadjar**
### Description

Documentation of `dicom-archive/get_dicom_info.pl` using perldoc.

### This PR resolves

- [x] Redmine ticket: https://redmine.cbrain.mcgill.ca/issues/15034

### To review

Read the documentation and make sure it all makes sense.

### CODE REVIEW

File: dicom-archive/get_dicom_info.pl [hunk 1]
```diff
@@ -160,15 +262,39 @@ sub get_dircos {
     return ($xcos, $ycos, $zcos);
 }
 
-# Routine to convert world coordinates
+=pod
+
+=head3 convert_coordinates(@coords)
+
+Routine to convert world coordinates (multiply them by -1).
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: Actually, only the `X` and `Y` part of the coordinates are multiplied by -1 and `Z` is left as is.

File: dicom-archive/get_dicom_info.pl [hunk 2]
```diff
@@ -160,15 +262,39 @@ sub get_dircos {
...
     $coords[1] *= -1;
     return @coords;
 }
 
-# Routine to compute a dot product
+=pod
+
+=head3 vector_dot_product($vec1, $vec2)
+
+Routine to compute a dot product.
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: I would rephrase this: Routine to compute the dot product of two vectors.

File: dicom-archive/get_dicom_info.pl [hunk 3]
```diff
@@ -190,18 +329,47 @@ sub vector_cross_product {
...
 
+=pod
+
+=head3 trim($input)
+
+Remove leading and trailing spaces from the $input variable
+
+INPUT: string to remove leading and trailing spaces from
+
+RESULTS: string without leading and trailing spaces
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: RESULTS => RETURNS

Comment (nicolasbrossard) [reply dicom-archive/get_dicom_info.pl]: Oh and I know it's not part of your PR but this method should be removed: there's already a Perl built-in function that does exactly this: `chomp`

File: dicom-archive/get_dicom_info.pl [hunk 4]
```diff
@@ -190,18 +329,47 @@ sub vector_cross_product {
...
     local($input) = @_;
     $input =~ s/^\s+//;
     $input =~ s/\s+$//;
     return $input;
 }
 
+=pod
+
+=head3 showcroft()
+
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: No doc here?

File: dicom-archive/get_dicom_info.pl [hunk 5]
```diff
@@ -190,18 +329,47 @@ sub vector_cross_product {
...
-# Routine to split a DICOM list into a perl list
+=pod
+
+=head3 split_dicom_list($dlist)
+
+Routine to split a DICOM list into a perl list
+
+INPUT: list of DICOMs
+
+RETURNS: string with the list of DICOM files
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: The functions adds `0` to each element so I am not sure it returns a list of DICOM **files**...It looks like it's returning a list of numbers.

File: dicom-archive/get_dicom_info.pl [hunk 6]
```diff
@@ -291,6 +478,19 @@ sub InfoOption {
     1;
 }
 
+=pod
+
+=head3 TwoArgInfoOption($option, $rest)
+
+Greps the group and element information from the GetOpt table options specified
+and check that the two arguments required by the option has been set.
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: ...and check**s** that the two arguments required by the option **have** been set

File: dicom-archive/get_dicom_info.pl (hunk 5 above)
Comment (cmadjar) [reply dicom-archive/get_dicom_info.pl]: You're right. Tried to make it clearer...

File: dicom-archive/get_dicom_info.pl (hunk 6 above)
Comment (cmadjar) [reply dicom-archive/get_dicom_info.pl]: done

File: dicom-archive/get_dicom_info.pl (hunk 4 above)
Comment (nicolasbrossard) [reply dicom-archive/get_dicom_info.pl]: Accessor for field `@croft` maybe?

File: dicom-archive/get_dicom_info.pl [hunk 7]
```diff
@@ -291,6 +478,19 @@ sub InfoOption {
     1;
 }
 
+=pod
+
+=head3 TwoArgInfoOption($option, $rest)
+
+Greps the group and element information from the GetOpt table options specified
+and check that the two arguments required by the option have been set.
```
Comment (nicolasbrossard) [inline dicom-archive/get_dicom_info.pl]: ...and check**s**...

Comment (cmadjar) [reply dicom-archive/get_dicom_info.pl]: done

**Comment by cmadjar** [comment]
@nicolasbrossard Thanks! Updated the PR with your comments :)
